### PR TITLE
Experimental v3 Procedural entities with depth and normals

### PR DESCRIPTION
**Pull request by SamGondelman**
Edit:  hmmm something seems to be wrong with transparent procedural entities...I'll fix that when I get a chance

Test plan:
- Perform these tests on various graphics cards: NVIDIA, AMD, test on Mac
- Run this from your console:
```
Entities.addEntity({
	type: "Box",
	position: Vec3.sum(MyAvatar.position, Vec3.multiply(4.0, Quat.getForward(MyAvatar.orientation))),
	dimensions: {x: 0.5, y: 0.5, z: 0.5},
	collisionless: true,
	userData: JSON.stringify({"ProceduralEntity":{"version":3,"shaderUrl":"https://gist.githubusercontent.com/SamGondelman/932c8c03fd39b0549f0a07c0a736abd9/raw/8460af956c640d7b73c0e30ed7a4212d9876c42d/raymarch.fs","uniforms":{"scene":0}}})
})
```
- You should see some grey spheres moving around a green rippling ball that morphs into a beach ball.  They should be lit correctly by the keylight.  Try placing other lights and entities around.  The spheres should be lit by the local lights and should depth-test properly with the objects.
- Look at the balls in VR.  They should appear 3D.  Pass your hand through them.  They should intersect your hand properly (with respect to depth, not physics).
- Run the procedural entity test: https://github.com/highfidelity/hifi_tests/blob/master/tests/content/entity/procedural/test.md
- Test any other procedural content you can find (the hexagon wall in Eschatology, etc.).  They should be unchanged.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11655/HighFidelity-Beta-PR13712-ce8420d.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11655/HighFidelity-Beta-PR13712-ce8420d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11655/HighFidelity-Beta-Interface-PR13712-ce8420d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11659/HighFidelity-Beta-PR13712-61755f2.apk

File: libraries/procedural/src/procedural/Procedural.cpp [hunk 1]
```diff
@@ -133,6 +133,11 @@ void Procedural::setProceduralData(const ProceduralData& proceduralData) {
     _dirty = true;
     _enabled = false;
 
+    if (proceduralData.version != _data.version) {
+        _data.version = proceduralData.version;
+        _shaderDirty = true;
+    }
```
Comment (SamGondelman) [inline libraries/procedural/src/procedural/Procedural.cpp]: This was missing, which means v1 shaders have probably been broken for a while.

File: libraries/render-utils/src/simple_fade.slf [hunk 2]
```diff
@@ -1,106 +0,0 @@
...
-                1.0,
-                diffuse,
-                max(0, 1.0 - shininess / 128.0),
-                length(specular),
-                DEFAULT_EMISSIVE+fadeEmissive,
-                DEFAULT_OCCLUSION,
-                DEFAULT_SCATTERING);
-        }
-    }
-}
```
Comment (SamGondelman) [inline libraries/render-utils/src/simple_fade.slf]: This was unused for now, removing it until we add entity fading and fix it on procedurals.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11659/HighFidelity-Beta-PR13712-61755f2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11659/HighFidelity-Beta-Interface-PR13712-61755f2.exe

**Comment by SamGondelman** [comment]
included in https://github.com/highfidelity/hifi/pull/13823

**Pull request closed without merging**